packages/core/__tests__/BackgroundProcessManager-test.ts

```diff
@@ -0,0 +1,629 @@
+import Observable from 'zen-observable-ts';
+import {
+	BackgroundProcessManager,
+	BackgroundProcessManagerState,
+} from '../src/Util/BackgroundProcessManager';
+
+describe('BackgroundProcessManager', () => {
+	test('can wait for a promise to finish', async () => {
+		// when we get passed `close()`, we need proof that the promise we're
+		// waiting on actually ran. <in a booming voice>: THIS IS THAT PROOF.
+		let proof = false;
+
+		const manager = new BackgroundProcessManager();
+
+		// add a job that finishes later, but don't wait for it.
+		// we want to ensure that close() is called before the promise
+		// completes.
+		manager.add(async () => {
+			return new Promise(resolve => {
+				setTimeout(() => {
+					proof = true;
+					resolve();
+				}, 50);
+			});
+		});
+
+		// the job should not have completed at the time we call this.
+		await manager.close();
+
+		expect(proof).toBe(true);
+	});
+
+	test('returns the value of the promise when adding it', async () => {
+		const manager = new BackgroundProcessManager();
+
+		const value = await manager.add(async () => Promise.resolve('VALUE'));
+
+		await manager.close();
+		expect(value).toEqual('VALUE');
+	});
+
+	test('passes thrown promise errors through', async () => {
+		const manager = new BackgroundProcessManager();
+
+		try {
+			await manager.add(async () => {
+				throw new Error('not today, friend!');
+			});
+			expect(true).toBe(false);
+		} catch (error) {
+			expect(error.message).toEqual('not today, friend!');
+		}
```

Comment:
```suggestion
    expect(
      manager.add(async () => {
        throw new Error('not today, friend!');
      })
    ).rejects.toThrow('not today, friend!');
```

Could use this in a few other places as well.

Reply:
OK. I'm not sure what I did wrong previously, but this works. I'll make a quick pass for patterns like this that are easy to update.